src/server/game/Combat/CombatManager.cpp

```diff
@@ -17,6 +17,33 @@
 
 #include "CombatManager.h"
 #include "Unit.h"
+#include "Player.h"
+
+/*static*/ bool CombatManager::IsCombatValid(Unit const* a, Unit const* b)
+{
+    // Checks combat validity before initial reference creation.
+    // For the combat to be valid...
+    // ...the two units need to be different
+    if (a == b)
+        return false;
+    // ...the two units need to be in the world
+    if (!a->IsInWorld() || !b->IsInWorld())
+        return false;
+    // ...the two units need to both be alive
+    if (!a->IsAlive() || !b->IsAlive())
+        return false;
+    // ...the two units need to be on the same map
+    if (a->GetMap() != b->GetMap())
+        return false;
+    // ...the two units need to be in the same phase
```

Comment:
> If any given raid member phases out, their threat is offlined, but maintained. When they re-enter the main phase, the threat still exists.
> Since the threat is only offlined, combat state between Algalon and players is maintained while the players are phased.

Thats exactly what I meant 😄